pkg/operator/ceph/cluster/controller.go

```diff
@@ -694,3 +753,48 @@ func ClusterOwnerRef(namespace, clusterID string) metav1.OwnerReference {
 		BlockOwnerDeletion: &blockOwner,
 	}
 }
+
+// GetAllStorageNodes returns an array of v1.Node objs this cluster has osds on
+func GetAllStorageNodes(context *clusterd.Context, namespace string) ([]v1.Node, error) {
+	// Generate the LabelSelector
+	discoveredNodes, err := discoverStorageNodes(context, namespace)
+	nodeNameList := make([]string, 0, 3)
+	for nodeName := range discoveredNodes {
+		nodeNameList = append(nodeNameList, nodeName)
+	}
+	labelSelector := fmt.Sprintf("kubernetes.io/hostname in (%v)", strings.Join(nodeNameList, ", "))
+
+	//get the node list
+	listOpts := metav1.ListOptions{LabelSelector: labelSelector}
+	nodeList, err := context.Clientset.Core().Nodes().List(listOpts)
+	return nodeList.Items, err
+}
+
+func discoverStorageNodes(context *clusterd.Context, namespace string) (map[string][]*apps.Deployment, error) {
```

Comment:
This looks like a duplicate method. Could we factor out the original code from `discoverStorageNodes()` in osd.go so that you can just call it from this package?

Reply:
+1